src/values/ray-query.cxx

```diff
@@ -0,0 +1,297 @@
+        return std::vector<float> {object_ray_dir.x, object_ray_dir.y, object_ray_dir.z};
+    }
+
+    /// @brief Get the committed or candidate intersection object-space ray origin.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return object-space ray origin.
+    glm::vec3 getIntersectionObjectRayOriginGLM(const unsigned intersection) const {
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionObjectRayOrigin(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate intersection object-space ray origin.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return object-space ray origin.
+    std::vector<float> getIntersectionObjectRayOrigin(const unsigned intersection) const {
+        glm::vec3 object_ray_origin = getIntersectionObjectRayOriginGLM(intersection);
+        return std::vector<float> {object_ray_origin.x, object_ray_origin.y, object_ray_origin.z};
+    }
+
+    /// @brief Get the committed or candidate intersection 3x4 object-to-world matrix.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return 4x3 world-to-object column-major order matrix.
+    glm::mat4x3 getIntersectionObjectToWorldGLM(const unsigned intersection) const {
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionObjectToWorld(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate intersection 3x4 object-to-world matrix.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return 4x3 world-to-object column-major order matrix.
+    std::vector<std::vector<float>> getIntersectionObjectToWorld(const unsigned intersection) const {
+        glm::mat4x3 object_to_world = getIntersectionObjectToWorldGLM(intersection);
+
+        std::vector<std::vector<float>> result;
+        for (unsigned col = 0; col < 4; ++col) {
+            result.push_back(std::vector<float> {object_to_world[col].x, object_to_world[col].y, object_to_world[col].z}
+            );
+        }
+
+        return result;
+    }
+
+    /// @brief Get the committed or candidate intersection 3x4 world-to-object matrix.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return 4x3 world-to-object column-major order matrix.
+    glm::mat4x3 getIntersectionWorldToObjectGLM(const unsigned intersection) const {
+        assert(intersection == 0 || intersection == 1);
+        return tlas.getIntersectionWorldToObject(static_cast<bool>(intersection));
+    }
+
+    /// @brief Get the committed or candidate intersection 3x4 world-to-object matrix.
+    /// @param intersection type of intersection: 0-candidate, 1-committed.
+    /// @return 4x3 world-to-object column-major order matrix.
+    std::vector<std::vector<float>> getIntersectionWorldToObject(const unsigned intersection) const {
```

Comment:
I see that you casted the input of getIntersectionWorldToObjectGLM to a bool, then the parameter for getIntersectionWorldToObject is an unsigned int again. I like that you casted it to a bool- please update the argument of getIntersectionWorldToObject.